upstairs/src/mend.rs

```diff
@@ -327,49 +181,20 @@ fn find_dest(
     c2: &RegionMetadata,
     log: &Logger,
 ) -> Vec<ClientId> {
-    let mut dest: Vec<ClientId> = Vec::new();
+    let c = ClientData::from_fn(|cid| {
+        [c0, c1, c2][cid.get() as usize].get(i).unwrap()
+    });
+    let s = c[source];
 
-    /*
-     * Put the three downstairs RegionMetadata structs into an array with the
-     * index being the client ID.
-     */
-    let rec = ClientData([c0, c1, c2]);
+    let out = ClientId::iter()
+        .filter(|&i| i != source && (s.dirty || c[i] != s))
+        .collect::<Vec<_>>();
 
     info!(
         log,
-        "find dest for source {} for extent at index {}", source, i
+        "find dest for source {source} for extent at index {i} => {out:?}"
```

Comment:
I think we can update this message to indicate we have found our result.  This was printing before we did the search, but now this log comes after we have found our answer.

Reply:
Sure, changed to `"found dest ..."`